spec/orm/activerecord_spec.rb

```diff
@@ -664,6 +663,33 @@ def filename
 
       end
 
+      describe '#changes' do
+        it "should be generated" do
+          @event.image = stub_file('test.jpeg')
+          expect(@event.changes).to eq({'image' => [nil, 'test.jpeg']})
+        end
+
+        it "shouldn't be generated when the attribute value is unchanged" do
+          @event.image = stub_file('test.jpeg')
+          @event.save!
+          @event.image = @event[:image]
+
+          expect(@event.changes).to be_blank
+          expect(@event).not_to be_changed
+          @event.remote_image_url = ""
+          expect(@event).not_to be_changed
+        end
+
+        it "shouldn't be generated after second save" do
+          @event.image = stub_file('old.jpeg')
+          @event.save!
+          @event.image = stub_file('new.jpeg')
+          @event.save!
+
+          expect(@event.changes).to be_blank
+          expect(@event).not_to be_changed
+        end
```

Comment:
The third test is the issue described in #2702 , Not sure yet whether we can make the second one pass.